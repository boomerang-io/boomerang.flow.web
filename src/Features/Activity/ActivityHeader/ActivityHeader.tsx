//@ts-nocheck
import React from "react";
import PropTypes from "prop-types";
import FeatureHeader from "Components/FeatureHeader";
import ActivityHeaderWidget from "./ActivityHeaderWidget";
import { SkeletonPlaceholder } from "@boomerang-io/carbon-addons-boomerang-react";
import { ArrowDownRight32, ArrowUpRight32 } from "@carbon/icons-react";
import styles from "./activityHeader.module.scss";

ActivityHeader.propTypes = {
  inProgressActivities: PropTypes.number,
  isError: PropTypes.bool,
  isLoading: PropTypes.bool.isRequired,
  failedActivities: PropTypes.number,
  runActivities: PropTypes.number,
  succeededActivities: PropTypes.number,
};

function ActivityHeader({
  inProgressActivities,
  isError,
  isLoading,
  runActivities,
  succeededActivities,
  failedActivities,
}) {
  const successRate = runActivities > 0 ? (succeededActivities + inProgressActivities) / runActivities : 0;
  const successRatePercentage = Math.round(successRate * 100);
  const emoji = successRatePercentage > 79 ? "ðŸ™Œ" : successRatePercentage > 49 ? "ðŸ˜®" : "ðŸ˜¨";

  return (
    <FeatureHeader includeBorder={false}>
      <div className={styles.container}>
        <hgroup>
          <p className={styles.subtitle}>This is all of the</p>
          <h1 className={styles.title}>Activity</h1>
        </hgroup>
        <section className={styles.content}>
          {isLoading ? (
            <SkeletonPlaceholder className={styles.summarySkeleton} />
          ) : isError ? (
            <>
              <p className={styles.text}>Today's numbers</p>
              <ActivityHeaderWidget text="Runs" value={"--"} />
              <ActivityHeaderWidget text="Successes" value={"--"} />
              <ActivityHeaderWidget text="Failures" value={"--"} />
              <ActivityHeaderWidget text="Success rate" value={"--"} />
            </>
          ) : (
            <>
              <p className={styles.text}>Today's numbers</p>
              <ActivityHeaderWidget icon={ArrowUpRight32} text="Runs" value={runActivities} />
              <ActivityHeaderWidget icon={ArrowUpRight32} text="Successes" value={succeededActivities} />
              <ActivityHeaderWidget icon={ArrowDownRight32} text="Failures" value={failedActivities} />
              <ActivityHeaderWidget icon={emoji} text="Success rate" value={`${successRatePercentage}%`} />
            </>
          )}
        </section>
      </div>
    </FeatureHeader>
  );
}

export default ActivityHeader;
