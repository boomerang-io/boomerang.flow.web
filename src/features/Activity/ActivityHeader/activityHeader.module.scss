@import "~@boomerang/boomerang-styles/lib/colors";

.container {
  background-color: $white;
  display: flex;
  height: 4.3125rem;
  justify-content: space-between;
  width: 100%;
}

.subtitle {
  color: $gray-90;
  font-size: 1.25rem;
  font-weight: 300;
  line-height: 1.5625rem;
}

.title {
  color: $gray-90;
  font-size: 2rem;
  font-weight: 600;
  line-height: 2.625rem;
}

.content {
  align-items: center;
  display: flex;
  margin: 2rem 2.125rem 1.5625rem 0;
}

.text {
  color: $gray-90;
  font-size: 1.25rem;
  font-weight: 300;
  line-height: 1.5625rem;
}

.summarySkeleton {
  width: 25rem;
  height: 4rem;
}
