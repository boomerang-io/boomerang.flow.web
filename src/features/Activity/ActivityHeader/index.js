import React from "react";
import PropTypes from "prop-types";
import FeatureHeader from "Components/FeatureHeader";
import ActivityHeaderWidget from "./ActivityHeaderWidtget";
import { SkeletonPlaceholder } from "carbon-components-react";
import { ArrowDownRight32, ArrowUpRight32 } from "@carbon/icons-react";
import styles from "./activityHeader.module.scss";

ActivityHeader.propTypes = {
  inProgressActivities: PropTypes.number.isRequired,
  isLoading: PropTypes.bool.isRequired,
  failedActivities: PropTypes.number,
  runActivities: PropTypes.number,
  succeededActivities: PropTypes.number
};

function ActivityHeader({ inProgressActivities, isLoading, runActivities, succeededActivities, failedActivities }) {
  const successRate = runActivities > 0 ? (succeededActivities + inProgressActivities) / runActivities : 0;
  const successRatePercentage = Math.round(successRate * 100);
  const emoji = successRatePercentage > 79 ? "ðŸ™Œ" : successRatePercentage > 49 ? "ðŸ˜®" : "ðŸ˜¨";

  return (
    <FeatureHeader includeBorder={false}>
      <div className={styles.container}>
        <section className={styles.headline}>
          <p className={styles.subtitle}>This is all of the</p>
          <p className={styles.title}>Activity</p>
        </section>
        <section className={styles.content}>
          {isLoading ? (
            <SkeletonPlaceholder className={styles.summarySkeleton} />
          ) : (
            <>
              <p className={styles.text}>Today's numbers</p>
              <ActivityHeaderWidget icon={ArrowUpRight32} text="Runs" value={runActivities} />
              <ActivityHeaderWidget icon={ArrowUpRight32} text="Successes" value={succeededActivities} />
              <ActivityHeaderWidget icon={ArrowDownRight32} text="Failures" value={failedActivities} />
              <ActivityHeaderWidget icon={emoji} text="Success rate" value={`${successRatePercentage}%`} />
            </>
          )}
        </section>
      </div>
    </FeatureHeader>
  );
}

export default ActivityHeader;
