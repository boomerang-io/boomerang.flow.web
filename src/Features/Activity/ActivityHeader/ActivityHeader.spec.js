import React from "react";
import ActivityHeader from "./index";
import { queryCaches } from "react-query";

afterEach(() => {
  queryCaches.forEach((queryCache) => queryCache.clear());
});

const props = {
  failedActivities: 10,
  runActivities: 25,
  succeededActivities: 13,
  inProgressActivities: 0,
  isLoading: false,
};

describe("ActivityHeader --- Snapshot", () => {
  it("Capturing Snapshot of ActivityHeader", () => {
    const { baseElement } = rtlRender(<ActivityHeader {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
