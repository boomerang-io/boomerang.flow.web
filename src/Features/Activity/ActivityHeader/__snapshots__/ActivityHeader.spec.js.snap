// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityHeader --- Snapshot Capturing Snapshot of ActivityHeader 1`] = `
<body
  id="app"
>
  <div>
    <header
      class="bx--bmrg-feature-header container"
    >
      <section
        class="bx--bmrg-feature-header__content"
      >
        <div
          class="bx--bmrg-feature-header__header"
        >
          <p
            class="bx--bmrg-feature-header-title"
          >
            This is all of the
          </p>
          <h1
            class="bx--bmrg-feature-header-title"
          >
            Activity
          </h1>
        </div>
      </section>
      <section
        class="content"
      >
        <p
          class="text"
        >
          Today's numbers
        </p>
        <div
          class="container"
        >
          <div
            class="valueAndIcon"
          >
            <p
              class="value"
            >
              25
            </p>
            <svg
              aria-label="Runs"
              class="icon"
              fill="currentColor"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 6L10 8 22.59 8 6 24.59 7.41 26 24 9.41 24 22 26 22 26 6 10 6z"
              />
            </svg>
          </div>
          <p
            class="text"
          >
            Runs
          </p>
        </div>
        <div
          class="container"
        >
          <div
            class="valueAndIcon"
          >
            <p
              class="value"
            >
              13
            </p>
            <svg
              aria-label="Successes"
              class="icon"
              fill="currentColor"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 6L10 8 22.59 8 6 24.59 7.41 26 24 9.41 24 22 26 22 26 6 10 6z"
              />
            </svg>
          </div>
          <p
            class="text"
          >
            Successes
          </p>
        </div>
        <div
          class="container"
        >
          <div
            class="valueAndIcon"
          >
            <p
              class="value"
            >
              10
            </p>
            <svg
              aria-label="Failures"
              class="icon"
              fill="currentColor"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 26L10 24 22.59 24 6 7.41 7.41 6 24 22.59 24 10 26 10 26 26 10 26z"
              />
            </svg>
          </div>
          <p
            class="text"
          >
            Failures
          </p>
        </div>
        <div
          class="container"
        >
          <div
            class="valueAndIcon"
          >
            <p
              class="value"
            >
              52%
            </p>
            <span
              aria-hidden="false"
              aria-label="Success rate"
              class="emoji"
              role="img"
            >
              ðŸ˜®
            </span>
          </div>
          <p
            class="text"
          >
            Success rate
          </p>
        </div>
      </section>
    </header>
  </div>
</body>
`;
